packages/runtime-core/src/componentProps.ts

```diff
@@ -122,10 +121,20 @@ type InferPropType<T> = [T] extends [null]
   ? U extends DateConstructor
     ? Date | InferPropType<U>
     : InferPropType<U>
-  : [T] extends [Prop<infer V, infer D>]
-  ? unknown extends V
-    ? IfAny<V, V, D>
-    : V
+  : [T] extends [{ type: infer U; default: infer D }]
+  ? [T] extends [PropOptions<infer V, infer E>]
+    ? V
+    : D
+  : [T] extends [{ type: infer U }]
+  ? [T] extends [PropOptions<infer V>]
+    ? V
+    : U
+  : [T] extends [{ default: infer U }]
+  ? [T] extends [PropOptions<infer V, infer D>]
+    ? D
+    : U
+  : [T] extends [PropType<infer V>]
+  ? V
```

Comment:
there's a few infers not being used can you remove any infers that are not being used?

Reply:
you're right, will do